src/legacy/core_plugins/metrics/server/lib/vis_data/helpers/annotations.js

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { last, head } from 'lodash';
+
+/**
+ * @param {Array} seriesGroup
+ * [
+ *  [
+ *    {
+ *      data: [
+ *        [1555189200000, 12],
+ *        [1555191100000, 42],
+ *        [1555263300000, 95],
+ *        ...coordinates,
+ *      ]
+ *      ...properties,
+ *    }
+ *    ...series,
+ *  ]
+ *  ...seriesGroups,
+ * ]
+ * @return {number} lastTimestamp
+ */
+export function getLastSeriesTimestamp(seriesGroup = []) {
+  let lastTimestamp = null;
+
+  seriesGroup.forEach(series => {
+    series.forEach(({ data }) => {
+      const dataLastTimestamp = head(last(data));
+
+      if (lastTimestamp === null) {
```

Comment:
looks like you can use `max `method from lodash. 

```
const dataLastTimestamp = head(last(data));
lastTimestamp  = _.max([lastTimestamp ,dataLastTimestamp])

```

optional: 
probably it make sense to use reduce instead of forEach 
`.reduce((lastTimestamp, {data}) => max(head([last(data), lastTimestamp]) )`

Reply:
Nice catch. Done